amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/BlockingRecordsPublisher.java

```diff
@@ -40,6 +42,7 @@
     private final GetRecordsRetrievalStrategy getRecordsRetrievalStrategy;
 
     private Subscriber<? super RecordsRetrieved> subscriber;
+    private final RequestDetails lastSuccessfulRequestDetails = new RequestDetails();
```

Comment:
Current implementation will print none for blocking record publisher implementation. let's update the requestId as part of getNextResults() method to keep this path inline with expectation.

Reply:
Yeah, I'll add this to `BlockingRecordsPublisher`.